aptos-move/aptos-vm-types/src/resolver.rs

```diff
@@ -53,6 +53,14 @@ pub trait TResourceView {
             .map(|maybe_state_value| maybe_state_value.map(StateValue::into_metadata))
     }
 
+    fn get_resource_state_value_size(
+        &self,
+        state_key: &Self::Key,
+    ) -> PartialVMResult<Option<usize>> {
+        self.get_resource_state_value(state_key, None)
+            .map(|maybe_state_value| maybe_state_value.map(|state_value| state_value.size()))
```

Comment:
Is this correct for groups?

Reply:
Great question. If it's not, we need to be group-aware when we are iterating through the write sets  for charging the gas and fees in the first place, which we are not today. @gelash ?

Reply:
yes we need to be resource group aware. 

op.materialized_size() and op.get_metadata_mut() is resource group aware, so current logic is correct.

maybe we should add op.get_previous_materialized_size(executor_view) to call appropriate method based on the op variant, and whether it is resource or resource group?

Reply:
@igor-aptos  I let GroupWrite carry a prev_group_size, PTAL
part of the reason is I don't have access to the resource group view trait from `op.prev_materialized_size()`